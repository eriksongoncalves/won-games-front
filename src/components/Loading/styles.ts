import styled, { css, keyframes } from 'styled-components';

const animate = keyframes`
  0% {
    height: 13px;
    width: 13px;
  }
  50% {
    height: 16px;
    width: 16px;
  }
  100% {
    height: 13px;
    width: 13px;
  }
`;

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 55px;
  margin: auto;
  color: ${({ theme }) => theme.colors.primary};
`;

export const Line = styled.div`
  ${() => css`
    width: 13px;
    height: 13px;
    border-radius: 50%;
    background-color: currentColor;
    display: inline-block;

    &:nth-child(1) {
      animation: ${animate} 1s 0s infinite ease-in-out;
    }
    &:nth-child(2) {
      animation: ${animate} 1s 200ms infinite ease-in-out;
    }
    &:nth-child(3) {
      animation: ${animate} 1s 400ms infinite ease-in-out;
    }
  `}
`;
