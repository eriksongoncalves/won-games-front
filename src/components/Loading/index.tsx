import * as S from './styles';

const Loading = () => (
  <S.Wrapper aria-label="loading...">
    <S.Line />
    <S.Line />
    <S.Line />
  </S.Wrapper>
);

export default Loading;
