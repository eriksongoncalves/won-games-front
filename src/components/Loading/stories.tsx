import { Story, Meta } from '@storybook/react/types-6-0';

import Loading from '.';

export default {
  title: 'Loading',
  component: Loading
} as Meta;

export const Default: Story = () => <Loading />;
