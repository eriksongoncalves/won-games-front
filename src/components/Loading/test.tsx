import { renderWithTheme } from 'utils/tests/helpers';

import Loading from '.';

describe('<Loading />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Loading />);

    expect(container).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
