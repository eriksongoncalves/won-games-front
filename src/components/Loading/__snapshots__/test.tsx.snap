// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 55px;
  margin: auto;
  color: #F231A5;
}

.c1 {
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background-color: currentColor;
  display: inline-block;
}

.c1:nth-child(1) {
  -webkit-animation: jeKoQt 1s 0s infinite ease-in-out;
  animation: jeKoQt 1s 0s infinite ease-in-out;
}

.c1:nth-child(2) {
  -webkit-animation: jeKoQt 1s 200ms infinite ease-in-out;
  animation: jeKoQt 1s 200ms infinite ease-in-out;
}

.c1:nth-child(3) {
  -webkit-animation: jeKoQt 1s 400ms infinite ease-in-out;
  animation: jeKoQt 1s 400ms infinite ease-in-out;
}

<div
  aria-label="loading..."
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c1"
  />
  <div
    class="c1"
  />
</div>
`;
